### DISCUSSION TITLE
Running Full windows or Wine64 with Boxedwine

### DISCUSSION THREAD

**Discussion by NarmakTwo** [danoon2/Boxedwine#98] [Ideas]
I've tested on a 32-bit Linux VM that it is possible to emulate 64-bit operating systems on a 32-bit system. This is done through QEMU and is probably possible with other emulators. Since I have zero knowledge with Emscripten and WASM, I don't think I'll be able to add this myself. Can anyone assist in writing  the code to be able to run Wine64 through Debian64 through QEMU? I know that it will be very hard on the performance but that isn't my main goal.

Any help would be appreciated.

Thanks,
Kamran

**Comment by danoon2** [comment]
I couldn't even imagine the performance of that :smile:  The web version is already way too slow.  It's been our goal to write a binary translator for wasm for many years to improve performance.

I also have doubts that Boxedwine could handle that.  The code just doesn't do that well on self modifying code.  I only recently started to get .NET to run.  I have yet to see Java run.

But even if you got it to run slowly, I've seen Wine behave badly when running super slow.  For instance, when running the slow CPU emulation (normal core) that the web version uses on my high end windows machine, the debug mode where I log all the CPU instructions, it slows it down so much that often Wine will time out or hit a really bad race condition where it will fail to create a window.

**Reply by NarmakTwo** [reply]
Also, what Linux emulator do you use for Boxedwine? I wonder if it's possible to run Anbox or Waydroid on it...

**Reply by danoon2** [reply]
It's actually my own emulation of all the kernal syscalls.  For the libc + other libs, I used Debian 10 and I'm currently experimenting with Tiny Core 15.

https://github.com/danoon2/Boxedwine/blob/84584f4d1ccab5f6a158cbbdf805d3ffb105cf7e/source/kernel/syscall.cpp#L1783

It's not a full Linux emulator, just the kernel.  So I don't start any services.
